### DISCUSSION TITLE
XML or PHP config files as default

### DISCUSSION THREAD

**Discussion by jazithedev** [symfony/symfony#42624] [Q&A] [answered]
Got a question. Just created a new Symfony project. Was adding some additional libraries like Doctrine, Doctrine Migrations, EasyAdminBundle etc. All of these generated configs in YAML. I would rather to have configs in XML or even in PHP to have proper syntax highlight in my PhpStorm IDE. So with that, I have two questions:

1. Is it possible to load XML/PHP configs by default (if available)?
2. Is it possible to automatically migrate current YAML configs onto XML/PHP?

I'm also asking about this because I saw [this](https://symfony.com/blog/new-in-symfony-5-3-config-builder-classes) news which mentioned, that Symfony team is moving towards to supporting PHP configuration by default. When creating a new project, I thought that it would be nice to have PHP configs from the beginning :).

Thank you in advance for any responses! Have a nice day!

**Comment by faizanakram99** [comment, marked as answer]
> Is it possible to load XML/PHP configs by default (if available)?

I believe if PHP will become defacto format for recipes, then yes they will be loaded automatically. Till then you would need to modify Kernel.php a bit.

For example this is our `Kernel.php` file to load both `yaml` as well as `php` config files. You can similarly add support for loading XML files as well, just replace `php,yaml` with `php,yaml,xml`

```php
protected function configureContainer(ContainerConfigurator $container): void 
{
        $container->import('../config/{packages}/*.{php,yaml}');
        $container->import("../config/{packages}/{$this->environment}/*.{php,yaml}");

        if (\is_file(\dirname(__DIR__).'/config/services.yaml')) {
            $container->import('../config/services.yaml');
            $container->import("../config/{services}_{$this->environment}.yaml");
        } else {
            $container->import('../config/{services}.php');
            $container->import("../config/{services}_{$this->environment}.php");
        }
    }
```

> Is it possible to automatically migrate current YAML configs onto XML/PHP?

Yes, you can use https://github.com/symplify/config-transformer package

Also, see this discussion https://github.com/symfony/symfony/discussions/42388

**Reply by jazithedev** [reply]
> Yes, you can use symplify/config-transformer package

So this package transform configs to PHP format. Is there any other which transforms YAML to XML?

**Reply by faizanakram99** [reply]
No it converts to all compatible formats, including XML, yaml, php and vice versa

**Reply by jazithedev** [reply]
Can you please tell me how to set it to convert formats to, for example, XML?

**Reply by faizanakram99** [reply]
Oh sorry, apparently the author ( @TomasVotruba ) removed support for conversion to other config formats in this PR https://github.com/symplify/symplify/pull/3247 

You can use `v9.3.15` in order to convert configs to XML format

`composer require symplify/config-transformer:9.3.15`

Docs of that version https://github.com/symplify/config-transformer/tree/v9.3.15

**Reply by jazithedev** [reply]
Thank you for the help :)
